src/main/java/com/zendesk/maxwell/schema/SchemaStore.java

```diff
@@ -67,6 +70,13 @@ public SchemaStore(Long serverID, CaseSensitivity sensitivity, Schema schema, Bi
 		this.position = position;
 	}
 
+	public SchemaStore createDerivedSchema(Schema newSchema, BinlogPosition position, List<ResolvedSchemaChange> deltas) throws SQLException {
+		if ( this.shouldSnapshotNextSchema )
+			return new SchemaStore(this.serverID, this.sensitivity, newSchema, position);
+		else
+			return new SchemaStore(this.serverID, this.sensitivity, newSchema, position, this.schema_id, deltas);
```

Comment:
was wondering if it was ok not to get sensitivity from the context, but the restored schema will get it from the context, so should be fine